src/createStreamingManager.ts

```diff
@@ -139,63 +139,71 @@ export async function createStreamingManager<T extends CreateStreamOptions>(
             }
-        } catch (e: any) {
-            callbacks.onError?.(e, { streamId: streamIdFromServer });
-        }
-    };
+        };
 
-    pcDataChannel.onmessage = (message: MessageEvent) => {
-        if (pcDataChannel.readyState === 'open') {
-            const [event, _] = message.data.split(':');
-            if (event === StreamEvents.StreamReady && !isConnected) {
-                onConnected();
+        pcDataChannel.onmessage = (message: MessageEvent) => {
+            if (pcDataChannel.readyState === 'open') {
+                const [event, _] = message.data.split(':');
+                if (event === StreamEvents.StreamReady && !isConnected) {
+                    onConnected();
+                }
             }
-        }
-    };
+        };
 
 
-    peerConnection.oniceconnectionstatechange = () => {
-        log('peerConnection.oniceconnectionstatechange => ' + peerConnection.iceConnectionState);
+        peerConnection.oniceconnectionstatechange = () => {
+            log('peerConnection.oniceconnectionstatechange => ' + peerConnection.iceConnectionState);
 
-        const newState = mapConnectionState(peerConnection.iceConnectionState);
+            const newState = mapConnectionState(peerConnection.iceConnectionState);
 
-        if (newState !== ConnectionState.Connected) {
-            callbacks.onConnectionStateChange?.(newState);
-        }
-    };
+            if (newState !== ConnectionState.Connected) {
+                callbacks.onConnectionStateChange?.(newState);
+            }
+        };
 
-    peerConnection.ontrack = (event: RTCTrackEvent) => {
-        log('peerConnection.ontrack', event);
-        callbacks.onSrcObjectReady?.(event.streams[0]);
-    };
+        peerConnection.ontrack = (event: RTCTrackEvent) => {
+            log('peerConnection.ontrack', event);
+            callbacks.onSrcObjectReady?.(event.streams[0]);
+        };
 
-    await peerConnection.setRemoteDescription(offer);
-    log('set remote description OK');
+        await peerConnection.setRemoteDescription(offer);
+        log('set remote description OK');
 
-    const sessionClientAnswer = await peerConnection.createAnswer();
-    log('create answer OK');
+        const sessionClientAnswer = await peerConnection.createAnswer();
+        log('create answer OK');
 
-    await peerConnection.setLocalDescription(sessionClientAnswer);
-    log('set local description OK');
+        await peerConnection.setLocalDescription(sessionClientAnswer);
+        log('set local description OK');
 
-    await startConnection(streamIdFromServer, sessionClientAnswer, session_id);
-    log('start connection OK');
+        await startConnection(streamIdFromServer, sessionClientAnswer, session_id);
+        log('start connection OK');
+    }
+
+    try {
+        await establishWebRtcConnection();
+    } catch {
+        await close(streamIdFromServer, session_id).catch(_ => { });
```

Comment:
Should we retry here? Or are we OK with the server holding the streaming until we reach the server-side timeout?